snow/networking/router/chain_router.go

```diff
@@ -63,6 +64,7 @@ type ChainRouter struct {
 	clock         mockable.Clock
 	log           logging.Logger
 	lock          sync.Mutex
+	closing       bool
```

Comment:
Do we care that the P-chain encountering a FATAL error could still cause the unexpected ERROR log? Since we are already FATALing... I don't think it would be too big of an issue... but just wanted to point out that it is still possible.

Reply:
I think if P-Chain fatals chain router won't be able to know it right, because `Push` won't return anything? Unless we talk about `SetOnStopped`/`removeChain` in this case it will be shutdown through node's shutdown path (through router's `onFatal` callback). I don't know if it violates any invariant if we shutdown if [here](https://github.com/ava-labs/avalanchego/blob/master/snow/networking/router/chain_router.go#L739-L741).

Reply:
Yeah... I think this is fine, if the node is FATALing we have bigger issues than this logging.